### PR TITLE
Enhance athena-start by Haru

### PR DESCRIPTION
**Pull request by xTararAisx**
### Added API Server Support and Enhanced Logging and Portability

- **Add API Server Variable**: Included the `A_SRV=api-server` variable to manage the API server alongside the existing login, char, and map servers.
- **Enhanced Process Checking**: Replaced `ps aux` with `pgrep` in the `isrunning` function to improve portability across different Unix-like systems.
- **Ensure Logs Directory Exists**: Added a check and creation step for the `logs` directory within the `do_start` function to ensure log files can be created without errors.
- **Redirect Output to Log Files**: Modified the `do_start` function to redirect the output of each server process to individual log files located in the `./logs` directory for easier debugging.
- **Include API Server in Stop Function**: Updated the `do_stop` function to include the API server, ensuring it gets properly terminated along with the other server components.
- **Include API Server in Status Function**: Updated the `do_status` function to include the API server, providing status checks for all server components.
- **Add API Server to Main Case Statement**: Modified the main case statement to include the API server in the start, stop, restart, and status commands.

Please review these changes and consider merging them into the main project. Thank you!

### CODE REVIEW

**Review by jasonch35 — CHANGES_REQUESTED** [review]

File: athena-start [hunk 1]
```diff
@@ -1,75 +1,214 @@
...
+	printf "${echo_herculeslogo}%-70s${echo_normal}\n" "               |  _  |/ _ \\ '__/ __| | | | |/ _ \\/ __|"
+	printf "${echo_herculeslogo}%-70s${echo_normal}\n" "               | | | |  __/ | | (__| |_| | |  __/\\__ \\"
+	printf "${echo_herculeslogo}%-70s${echo_normal}\n" "               \\_| |_/\\___|_|  \\___|\\__,_|_|\\___||___/"
+	printf "${echo_herculeslogo}%-70s${echo_normal}\n" ""
+	printf "${echo_herculeslogo}%-70s${echo_normal}\n" "                    http://hercules.ws/board/"
+	printf "${echo_herculeslogo}%-70s${echo_normal}\n" ""
+	if [ -d .git ] && type git >/dev/null 2>&1; then
+		printf "%-15s '${echo_bold}%s${echo_normal}'\n" "Git Hash:" "$(git rev-parse HEAD)"
+	fi
+	printf "${echo_bold_green}%-15s${echo_normal} ${echo_bold}http://hercules.ws/${echo_normal}\n" "Website/Forum:"
```
[the remark this answers is not in the document]
Comment (xTararAisx) [reply athena-start:102]: ups! done thing! :)

**Comment by jasonch35** [comment]
I used this for 3 days now. Here are my comments:

1. Herc already has `log` directory, can we save it there instead of making new `logs`?
2. Error, debug, and warning show messages are not showing as it should like Haru said
3. Maybe rename to `herc-start`? 😄

**Comment by skyleo** [comment]
> Thanks for the update! Ive fixed it and enhance it even further
> 
>     * Add optional arguments `--display-error-off` and `--display-error-on` to control error display.
> 
>     * Set error display to ON by default.
> 
>     * Update usage message to reflect new arguments.
> 
>     * Use existing Hercules log directory (`../log`) instead of creating a new one.
> 
>     * Clean and fix issues in `do_start`, `do_stop`, and `do_status` functions.
> 
>     * Ensure proper termination and status checking of processes.
> 
>     * Improve argument parsing for better user experience.

Those are really nice points. Perfect for individual commits using exactly these messages. Please use atomic commits. If you have to make a bullet list to describe the changes of one commit then it's meant to be multiple commits.

**Comment by xTararAisx** [comment]
huh? what?

**Comment by guilherme-gm** [comment]
What skyleo mentioned is that having huge commits doing lots of things at once is not a good idea. You listed several additions, they would be better being separate commits, so your changes are built step by step. Easier to review, easier to find bugs now and in the future.

Smaller commits allows us to see things like: I am reading a commit that says "I added X", and could think "Uh, why is it doing Y? -- oh this is a bug!"; in the future it would also be easier if we are wondering why something is there.

Right now, if we look at the commit history of your PR, we see:
- Update athena-start
- - Update athena-start
- - Update athena-start

I think those links may help you get started with rebasin

**Comment by guilherme-gm** [comment]
What skyleo mentioned is that having huge commits doing lots of things at once is not a good idea. You listed several additions, they would be better being separate commits, so your changes are built step by step. Easier to review, easier to find bugs now and in the future.

Smaller commits allows us to see things like: I am reading a commit that says "I added X", and could think "Uh, why is it doing Y? -- oh this is a bug!"; in the future it would also be easier if we are wondering why something is there.

Right now, if we look at the commit history of your PR, we see:
- Update athena-start
   - lots of things happened here
- Update athena-start
   - lots of things happened here
- Update athena-start
   - lots of things happened here
- ...

We have many changes coming at once, even if they are detailed in the commit body, they seem like a lot of unrelated changes happening at once. It would be better to have them as separate commits, like:

```
commit 1: Add optional arguments `--display-error-off` and `--display-error-on` to control error display.
commit 2: Set error display to ON by default.
commit 3: Update usage message to reflect new arguments.
commit 4: Use existing Hercules log directory (`../log`) instead of creating a new one.
// ...
```

I think those links may help you get started with a bit more info about why we are asking that and getting started with rebasing to split commits:
- https://dev.to/karaluton/a-guide-to-perfecting-pull-requests-2b66
- https://github.com/supercollider/supercollider/wiki/Rebasing-a-PR#rearranging-combining-commits-or-editing-commit-messages-with-rebase

**Pull request closed without merging**